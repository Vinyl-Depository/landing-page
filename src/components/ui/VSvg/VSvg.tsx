import React, { CSSProperties } from 'react';

import icons from 'src/icons';

import VSvgView from './VSvg.view';

interface IProps {
	readonly name: keyof typeof icons;
	readonly className?: string;
	readonly style?: CSSProperties;
	readonly onClick?: () => void;
}

const VSvg: React.FC<IProps> = (props: React.PropsWithChildren<IProps>) => {
	return (
		<VSvgView style={props.style} className={props.className} name={props.name} onClick={props.onClick} />
	);
};

VSvg.displayName = 'VSvg';
VSvg.defaultProps = {};

export default React.memo(VSvg);
