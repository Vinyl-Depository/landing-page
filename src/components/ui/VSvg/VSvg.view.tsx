import React, { CSSProperties } from 'react';

import icons from 'src/icons';

import classes from './VSvg.module.scss';

interface Props {
	readonly name: keyof typeof icons;
	readonly className?: string;
	readonly style?: CSSProperties;
	readonly onClick?: () => void;
}

const VSvgView: React.FC<Props> = (props: React.PropsWithChildren<Props>) => {
	const svgClasses = classes['container'] + (props.className ? ` ${props.className}` : '');

	return (
		<svg
			style={props.style}
			className={svgClasses}
			version="1.1"
			xmlnsXlink="http://www.w3.org/1999/xlink"
			xmlns="http://www.w3.org/2000/svg"
			viewBox={'0 0 ' + icons[props.name][0]}
			dangerouslySetInnerHTML={{ __html: icons[props.name][1] ?? '' }}
			onClick={props.onClick}
		/>
	);
};

VSvgView.displayName = 'VSvgView';
VSvgView.defaultProps = {};

export default VSvgView;
