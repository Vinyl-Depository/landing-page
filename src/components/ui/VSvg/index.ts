import VSvg from './VSvg';

export default VSvg;
