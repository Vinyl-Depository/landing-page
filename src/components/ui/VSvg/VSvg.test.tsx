import React from 'react';
import { render } from '@testing-library/react';

import VSvg from './VSvg';

describe('<VSvg>', () => {
	it('Should render the component unchanged', () => {
		const { container } = render(<VSvg name="carouselArrow" />);

		expect(container).toMatchSnapshot();
	});
});
