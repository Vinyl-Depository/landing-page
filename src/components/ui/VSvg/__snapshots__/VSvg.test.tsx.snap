// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VSvg> Should render the component unchanged 1`] = `
<div>
  <svg
    class="container"
    version="1.1"
    viewBox="0 0 31 54"
    xmlns="http://www.w3.org/2000/svg"
    xmlns:xlink="http://www.w3.org/1999/xlink"
  >
    
			
    <path
      d="m0 48.807 4.335 4.336L30.767 26.57 4.335 0 0 4.335l22.236 22.236L0 48.807Z"
    />
    
		
  </svg>
</div>
`;
